api.go

```diff
@@ -651,11 +651,11 @@ func (c Client) newRequest(method string, metadata requestMetadata) (req *http.R
 		return req, nil
 	} // Sign the request for all authenticated requests.
 
-	if c.signature.isV2() {
+	switch {
+	case c.signature.isV2():
 		// Add signature version '2' authorization header.
 		req = s3signer.SignV2(*req, c.accessKeyID, c.secretAccessKey)
-	} else if c.signature.isV4() || c.signature.isStreamingV4() &&
-		method != "PUT" {
+	case c.signature.isV4():
```

Comment:
> I reviewed the code and saw that StreamingSignature is enabled only for PUT request. Hence this code was not necessary. Let me know if I am reading anything incorrectly.

yes you are right, a request is stream v4 only when we are going to upload an object or a part, thanks.